core/pkg/fixhandler/testdata/inserts/tc-12-00-begin-with-document-separator.yaml

```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: begin-with-document-separator
+
+spec:
+  containers:
+  - name: nginx_container
+    image: nginx
```

Comment:
praise: great decision to add a new test case file!

This decision is good in particular because:
- it follows the existing test conventions for this functionality
- the existing conventions test our public-facing behavior and contract: that we take in a specific file, apply some fixes and return some specific expected file.

So this new test file will do the best thing tests are supposed to do: check that we keep the promises we make to the public.

Now, you might notice I made some changes to the PR. Among these changes is that I moved the calls to the sanitizer functions to the `ApplyFixToContent` function. This is because our aforementioned public-facing behavior is provided by the `ApplyFixToContent` function.

So if the `ApplyFixToContent` function promises the following contract:

> given a valid YAML file and a YAML expression, I will apply the provided fixes to it.

it only makes sense that this function would include the sanitization behavior in itself. Now, we just need to make sure that we test the case of YAML files that start with the document separator. I removed the tests of the unexported (private) sanitizer functions for this reason as well. There is no reason to test the brittle details of the implementation as long as we test the public contracts.

So the takeaway here is that the best practice is to test your contracts aka behavior that you promise to uphold to the clients.

Reply:
@XDRAGON2002 
> Since we already have test files in the `hybrid` test data folder now, I guess we can go ahead and remove the files I had added to the `insert` test data folder?

The cases are different, so let’s keep the one you’ve added. The more meaningful cases, the better.